src/tomato_bronkhorst/__init__.py

```diff
@@ -0,0 +1,405 @@
+            coords=[timestamps],
+            dims=["time"],
+            name="temperature",
+            attrs={"units": "°C"},
+        )
+
+        flow_rate_array = xr.DataArray(
+            flow_rate_data,
+            coords=[timestamps],
+            dims=["time"],
+            name="flow_rate",
+            attrs={"units": self.max_flow_unit},
+        )
+        pressure_array = xr.DataArray(
+            pressure_data,
+            coords=[timestamps],
+            dims=["time"],
+            name="pressure",
+            attrs={"units": "bar"},
+        )
+        dataset = xr.Dataset(
+            {
+                "temperature": temperature_array,
+                "flow_rate": flow_rate_array,
+                "pressure": pressure_array,
+            }
+        )
+
+        return dataset
+
+        def _set_attr(self, attr: str, val: Any, **kwargs: dict):
+            if attr in property_map:
+                params = self.property_map_instance._get_property(attr)
+                if "param_id" in params:
+                    self.instrument.writeParameter(params["param_id"], val)
+                else:
+                    self.instrument.write_parameters([{**params, "data": val}])
+            else:
+                raise ValueError(f"Unknown property: {attr}")
+
+        def _get_attr(self, attr: str, **kwargs: dict):
+            """
+            Retrieves the value of an attribute from the instrument.
+
+            Args:
+                attr (str): The name of the attribute to retrieve.
+
+            Returns:
+                The value of the requested attribute.
+            """
+
+        property_details = self.property_map_instance._get_property(attr)
+        if property_details:
+            if "param_id" in property_details:
+                return self.instrument.readParameter(property_details["param_id"])
+            else:
+                values = self.instrument.read_parameters([property_details])
+                return values[0]["data"]
+        else:
+            raise ValueError(f"Unknown property: {attr}")
+
+        def _attrs(self, **kwargs) -> dict[str, Attr]:
+            """
+            Returns a dictionary of available attributes for the device.
+
+            Returns:
+                dict: A dictionary of attribute names and their respective metadata.
+            """
+            attrs_dict = {
+                "temperature": Attr(type=float, units="Celsius"),
+                # as mentionned now, the user can not modify the data, only read them.
+                "flow": Attr(type=float, units=self.flow_units, rw=False, status=True),
+                "pressure": Attr(type=float, units=self.pressure_unit_id, rw=False),
+            }
+            if self.device_type == "PC":
+                attrs_dict["pressure"] = Attr(
+                    type=float, units=self.pressure_units, rw=False
+                )
+            return attrs_dict
+
+        def _capabilities(self, **kwargs) -> set:
```

Comment:
This should also be on the DeviceManager, and `capabilities` not `_capabilities`.